test/integration/setup/index.ts

```diff
@@ -147,6 +150,52 @@ export class TestEndtoEndCoordinator implements TestCoordinator {
       await proposal.simulate();
     }
 
+    if (config.category === ProposalCategory.DEBUG) {
+      console.log('Simulating DAO proposal in DEBUG mode (step by step)...');
+      console.log('  Title: ', config.proposal.title);
+
+      const signer = await getImpersonatedSigner(contracts.feiDAOTimelock.address);
+      await forceEth(contracts.feiDAOTimelock.address);
+
+      let totalGasUsed = 0;
+      for (let i = 0; i < config.proposal.commands.length; i++) {
```

Comment:
I find this logic in lines 161 - 194 very hard to follow. 

It's some form of parser for a DAO proposal? I would encapsulate into it's own function with a description of what it does and perhaps include an example input and output. 

If it's significantly extending e2e test functionality, I'd probably break out into a seperate PR